lib/prefetch.js

```diff
@@ -110,7 +110,7 @@ if (hasServiceWorkerSupport()) {
 
 function getPrefetchUrl (href) {
   let { pathname } = urlParse(href)
-  const url = `/_next/${__NEXT_DATA__.buildId}/pages${pathname}`
+  const url = `/_next/${__NEXT_DATA__.buildId}/${__NEXT_DATA__.pagesDirectory}${pathname}`
```

Comment:
I quite don't like this change.
We should not need to any changes to the client code.

For that, keep the pages in the `.next` as `pages`. There's no need for changing that.